modules/api/src/main/java/org/apache/ignite/compute/NodesJobTarget.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.compute;
+
+import java.util.Objects;
+import java.util.Set;
+import org.apache.ignite.network.ClusterNode;
+
+/**
+ * Nodes-based job execution target.
+ */
+public class NodesJobTarget implements JobTarget {
```

Comment:
the name seems to vague... What if we decide to introduce new target with semantic "all nodes", how we will name the container? Does it make sense to rename this to `AnyNodeTarget` to better reflect semantic? This it part of public API, thus we should choose naming carefully

Reply:
Good point, renamed to `AnyNodeJobTarget`. Also renamed the other one to `ColocatedJobTarget`.